### PR TITLE
Install EFA drivers

### PR DESCRIPTION
**Pull request by sean-smith**
Installs the EFA drivers. See [1]

[1] https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/efa-start.html#efa-start-enable

Signed-off-by: Sean Smith <seaam@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: recipes/_efa_install.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    mode '0644'
+    retries 3
+    retry_delay 5
+    not_if { ::File.exist?(efa_installer) }
+  end
+
+  bash "install efa" do
+    cwd node['cfncluster']['sources_dir']
+    code <<-NODE
+      yum remove -y openmpi
```
Comment (enrico-usai) [inline recipes/_efa_install.rb]: Could you add a comment describing why we need to remove openmpi?

Comment (sean-smith) [reply recipes/_efa_install.rb]: Added

File: recipes/_efa_install.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    not_if { ::File.exist?(efa_installer) }
+  end
+
+  bash "install efa" do
+    cwd node['cfncluster']['sources_dir']
+    code <<-NODE
+      # default openmpi installation conflicts with new install
+      # new one is installed in /opt/amazon/efa/bin/
+      yum remove -y openmpi
+      tar -xzf aws-efa-installer-latest.tar.gz
```
Comment (lukeseawalker) [inline recipes/_efa_install.rb]: use #{efa_installer} instead. Minor: rename the var efa_installer efa_tarball to be consistent with the other tarballs

Comment (sean-smith) [reply recipes/_efa_install.rb]: done

File: recipes/base_install.rb [hunk 3]
```diff
@@ -235,3 +235,6 @@
 
 # Install FSx options
 include_recipe "aws-parallelcluster::_lustre_install"
+
+# Install EFA
+include_recipe "aws-parallelcluster::_efa_install"
```
Comment (lukeseawalker) [inline recipes/base_install.rb]: just for consistency I would move the OS check here like we do for the fsx recipe

File: attributes/default.rb [hunk 4]
```diff
@@ -45,6 +45,8 @@
 default['cfncluster']['nvidia']['enabled'] = 'no'
 default['cfncluster']['nvidia']['driver_url'] = 'http://download.nvidia.com/XFree86/Linux-x86_64/418.56/NVIDIA-Linux-x86_64-418.56.run'
 default['cfncluster']['nvidia']['cuda_url'] = 'https://developer.nvidia.com/compute/cuda/10.0/Prod/local_installers/cuda_10.0.130_410.48_linux'
+# EFA
+default['cfncluster']['efa']['installer_url'] = 'https://s3-us-west-2.amazonaws.com/aws-efa-installer/aws-efa-installer-latest.tar.gz'
```
Comment (lukeseawalker) [inline attributes/default.rb]: we need to understand the implication of having "us-west-2"

Comment (sean-smith) [reply attributes/default.rb]: yep, china regions won't work we're working on resolving this

File: recipes/base_install.rb (hunk 3 above)
Comment (sean-smith) [reply recipes/base_install.rb]: I don't understand what you mean? FSx recipe checks the os in the `base_config.rb`, not in `base_install.rb`. We install efa drivers in `base_install.rb` so it'll be built into the ami, or installed at runtime for a custom ami.

Comment (lukeseawalker) [reply recipes/base_install.rb]: yeah, I meant to move the check
`if (node['platform'] == 'centos' && node['platform_version'].to_i >= 7) || node['platform'] == 'amazon'`
outside the recipe _efa_install, so it's not even called

`if (node['platform'] == 'centos' && node['platform_version'].to_i >= 7) || node['platform'] == 'amazon'
  include_recipe "aws-parallelcluster::_efa_install"
end
`

**Pull request merged**